### PR TITLE
modify rule to find best url to use for video

### PR DESCRIPTION
**Pull request by bhsiao55**

### CODE REVIEW

**Comment by bhsiao55** [comment]
@arunkarnati this might work.

File: src/ArticleDocumentMarshaler.php [hunk 1]
```diff
@@ -1119,14 +1119,52 @@ protected function transformVideoBlock(Message $block, array &$context): ?Compon
...
-     * @return string
-     */
     protected function getVideoUrl(Message $video): ?string
     {
-        return $video->get('kaltura_mp4_url');
+        if ($video->has('live_m3u8_url')) {
+            $videoUrl = $video->get('live_m3u8_url');
+        } elseif ($video->has('mezzanine_ref')) {
+            $mezzanineId = AssetId::fromString($video->get('mezzanine_ref')->getId());
+            $pathInfo = pathinfo($this->urlProvider->getUrl($mezzanineId));
```
Comment (gdbrown) [inline src/ArticleDocumentMarshaler.php:1128]: @arunkarnati let's actually move this to site level so we can use the Url provider from ovp to generate the url without having to make apple news dependent upon ovp

**Comment by arunkarnati** [comment]
@gdbrown - done, please check again. I will add the `mezzanine_ref` URL resolution to the site level marshalers.

**Comment by arunkarnati** [comment]
> @arunkarnati i'd put that whole business as the site level. getVideoUrl really ought to be abstract function

so, no `getVideoURL` support at the Triniti level, right? I would like to know why it isn't a good idea to have the basic support as we did for notifier payloads?

**Comment by gdbrown** [comment]
@arunkarnati leave the current one htere... the kaltura_mp4_url, that's fine for now.  at some point the "source" urls might be used.  however, digging into mezz refs, kaltura flavors is highly specific to a site so it's hard to make it generic.

**Comment by arunkarnati** [comment]
I will close this PR and move the code to the site-level.

**Pull request closed without merging**